modules/mod_ginger_rdf/support/ginger_json_ld.erl

```diff
@@ -239,25 +240,36 @@ compact_predicate(Predicate) ->
 -spec deserialize(tuple() | list()) -> #rdf_resource{}.
 deserialize(JsonLd) when is_map(JsonLd) ->
     Context = maps:get(<<"@context">>, JsonLd, #{}),
+    deserialize_with_context(undefined, JsonLd, Context);
+deserialize(JsonLd) ->
+    %% Fall back to mochijson {struct, ...} structure
+    open(JsonLd).
+
+%% Guide the parser; indicate that we are actually looking for a specific id
+%% so we don't get confused when we encounter multiple subjects in the jsonld.
+deserialize(Id, JsonLd) when is_map(JsonLd) ->
+    Context = maps:get(<<"@context">>, JsonLd, #{}),
+    deserialize_with_context(Id, JsonLd, Context).
```

Comment:
Nitpick, but isn't this the same as:
```suggestion
deserialize(Id, JsonLd) when is_map(JsonLd) ->
    deserialize_with_context(Id, JsonLd, #{}).
```
given that `deserialize_with_context` fetches the local context too?

Reply:
Agreed, this applies to both function clauses (also the one at :241).

Reply:
Applied the suggestion to both clauses.